### ISSUE TITLE
[Windows] Dart application code should be built into app.dll

### ISSUE THREAD

**Issue by cbracken** [flutter/flutter#106844]
Today, we build the Dart AOT application code into the app executable on Windows. On macOS and iOS, we build that Dart application into its own App.framework which we dynamically load at runtime, then resolve symbols to get the executable AOT code. We should do the same for Windows.

This is a prerequisite for add-to-app scenarios where Flutter is being added (as a library) to an existing application rather than being built as its own application. Examples include:
* An existing windows app that wants to add additional views/windows implemented using Flutter
* A plugin for a commercial app that supports plugins and wants to implement the plugin UI in Flutter

**Comment by cbracken** [comment]
TL;DR This is actually already implemented and we'd forgotten.

* During `flutter assemble` on Windows, we build an `app.so` ELF binary.
* The [`WindowsAotBundle`](https://github.com/flutter/flutter/blob/c6d44e25fbc4947cf177657118f3e6154371969c/packages/flutter_tools/lib/src/build_system/targets/windows.dart#L159-L195) rule copies it to the bundle directory.
* At runtime the [`DartProject` constructor](https://github.com/flutter/engine/blob/ce062511629433489047417aea35de1a9483e87e/shell/platform/windows/client_wrapper/include/flutter/dart_project.h#L20) takes an `aot_library_path` or alternatively [defaults](https://github.com/flutter/engine/blob/ce062511629433489047417aea35de1a9483e87e/shell/platform/windows/client_wrapper/include/flutter/dart_project.h#L43) to the bundle plath + `\app.so`.
* At runtime in AOT mode builds, we [load that AOT data](https://github.com/flutter/engine/blob/ce062511629433489047417aea35de1a9483e87e/shell/platform/windows/flutter_windows_engine.cc#L217-L223) in `FlutterWindowsEngine` and [stuff it into the `FlutterProjectArgs`](https://github.com/flutter/engine/blob/ce062511629433489047417aea35de1a9483e87e/shell/platform/windows/flutter_windows_engine.cc#L334-L336) which we [pass to `FlutterEngineRun`](https://github.com/flutter/engine/blob/ce062511629433489047417aea35de1a9483e87e/shell/platform/windows/flutter_windows_engine.cc#L342-L343).
